### PR TITLE
remove leading, trailing whitespace and common indentation from strings

### LINKED ISSUE #51: remove common indentation from strings and so forth
At https://dada-lang.org/dyn_tutorial/hello.html, the tutorial claims:

> Strings in Dada can spread over multiple lines. Leading and trailing whitespace is stripped by default, and we also remove any common indentation from each line.

But this is clearly not true. =) We should fix this! I imagine this would take place in the parser? Or perhaps the validation engine, when it lowers format strings. Somewhere like that.

### PR DESCRIPTION
**Pull request by XFFXFF**
Fix #51

### CODE REVIEW

**Review by nikomatsakis — CHANGES_REQUESTED** [review]
This is awesome! One suggestion (see below) to improve. Thoughts?

File: components/dada-validate/src/validate/validator.rs [hunk 1]
```diff
@@ -535,3 +539,33 @@ impl<'me> Validator<'me> {
...
+        return s.to_string();
+    }
+    let non_empty_lines = s
+        .lines()
+        .filter(|&line| !line.trim().is_empty())
+        .collect::<Vec<&str>>();
+
+    let indent = non_empty_lines
+        .iter()
+        .map(|line| line.chars().into_iter().take_while(|c| *c == ' ').count())
```
[the remark this answers is not in the document]
Comment (XFFXFF) [reply components/dada-validate/src/validate/validator.rs:555]: I get it. I might fix it tomorrow. :smile:

[the remark this answers is not in the document]
Comment (nikomatsakis) [reply components/dada-validate/src/validate/validator.rs:555]: I'm going to land this PR in the meantime :)

**Pull request merged**